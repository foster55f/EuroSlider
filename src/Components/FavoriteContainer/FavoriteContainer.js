import React from 'react'
import HighlightCard from '../HighlightCard/HighlightCard';
import { connect } from 'react-redux';
import { favoriteGames } from '../../reducers/favoriteGames';


export const FavoriteContainer = ({ games, displayGames, favoriteGames }) => {
    return (
        <div className= 'game-container'>
            {favoriteGames.map(game => {
            return (
                <HighlightCard
                key={game.competition.id}
                id={game.competition.id} 
                title={game.title}
                image={game.thumbnail}
                url={game.url}
                embed ={game.videos[0].embed}     
            />
            )
        })}
        </div>
    )
}
export const mapStateToProps = state => ({
    games: state.games,
    displayGames: state.displayGames,
    favoriteGames: state.favoriteGames
  })
  
  export default connect(mapStateToProps)(FavoriteContainer)