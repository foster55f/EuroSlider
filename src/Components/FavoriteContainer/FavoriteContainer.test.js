import React from 'react';
import { shallow } from 'enzyme';
import { FavoriteContainer, mapStateToProps } from './FavoriteContainer';

describe('FavoriteContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FavoriteContainer
        favoriteGames={[
          
      ]}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

  describe('mapsStateToProps', () => {
    it('should return only games and favorite games from the store', () => {
        const mockState = {
            games: [],
            displayGames: [],
            favoriteGames: [],
        };
        const expected = {
            games: [],
            displayGames: [],
            favoriteGames: [],
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
        });
    });
 