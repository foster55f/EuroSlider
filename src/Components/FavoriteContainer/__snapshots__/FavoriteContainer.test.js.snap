// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteContainer should match the snapshot 1`] = `ShallowWrapper {}`;
